Updated stripeclient snippets in Readme.md (#3137)
stripe/stripe-dotnet: README.md

## Usage

### Using StripeClient

In version 46 of the Stripe .NET SDK, we have enhanced the `StripeClient` class to be the entry point to access all services that had to be previously independently instantiated with global configuration. This improves discoverability during IDE auto-completion and creates a more intuitive developer experience for you. 

Each client instantiation can have its own configuration so you can access Stripe API with different API keys or different configuration (like number of retries) on a per client basis and without changing a global configuration.

```C#
// StripeClient pattern (Recommended)
var client = new StripeClient("sk_test_...");
Customer customer = client.V1.Customers.Get("cus_1234");

// Global Configuration pattern (Legacy)
StripeConfiguration.ApiKey = "sk_test_...";
var service = new CustomerService();
Customer customer = service.Get("cus_1234");
```

> The traditional Global Configuration pattern will continue to work for everyone. 

Reviewer: ```suggestion
The previous global configuration pattern will continue to be supported.
```

## Usage

> ### Using StripeClient

Author: @ramya-stripe Can you please take a look at this?

Stripe .NET SDK offers a more flexible approach of working with the Stripe API through the `StripeClient` class. 
StripeClient pattern has several advantages over the traditional approach of instantiating services with a global configuration: 
- Client based configuration: Allows you to instantiate several clients with different configurations(API key, network retries etc.). 

> - Single Root Entry Point: The StripeClient provides a centralized entry point to all available Stripe services. This improves discoverability through IDE auto-completion and creates a more intuitive developer experience for you.

Reviewer: @jar-stripe Does the above read right to you?

> - Single Root Entry Point: The StripeClient provides a centralized entry point to all available Stripe services. This improves discoverability through IDE auto-completion and creates a more intuitive developer experience for you.

Reviewer: done

    Email = "customer@example.com"
};

var client = new StripeClient("sk_test_...");

> var service = client.V1.Customers;

Author: This is to directly replicate what a user will see in API reference to avoid confusion when jumping from readme to apiref. 
One liner is simpler but it would require a refactor of .NET examples.

> var service = client.V1.Customers;

Reviewer 2: Hm, ok. Do we have that on our radar?

> var service = client.V1.Customers;

Author: Not planned atm. Might be easier to do when we drop support for resources.

> var service = client.V1.Customers;

Reviewer: Add a new section "Using StripeClient" and talk about the benefits while maintaining that the old way of instantiating services with global config will still work

Advantages to call out
- No global config - you can create multiple clients with different configuration
- A single root entry point from where you can discover all available services
